src/modules/networks/networks-store.background.ts

```diff
@@ -1,16 +1,99 @@
+import { client } from 'defi-sdk';
+import type { WalletRecord } from 'src/shared/types/WalletRecord';
 import { chainConfigStore } from '../ethereum/chains/ChainConfigStore';
+import { configureBackgroundTestClient } from '../defi-sdk/background';
+import type { ChainId } from '../ethereum/transactions/ChainId';
 import { NetworksStore } from './networks-store';
+import type { NetworkConfig } from './NetworkConfig';
+import type { Chain } from './Chain';
 
-export const networksStore = new NetworksStore(
+export const mainNetworksStore = new NetworksStore(
   { networks: null },
   {
     getEthereumChainConfigs: async () => {
       await chainConfigStore.ready();
       return chainConfigStore.getState().ethereumChainConfigs;
     },
+    client,
+    testnetMode: false,
   }
 );
 
+export const testenvNetworksStore = new NetworksStore(
+  { networks: null },
+  {
+    getEthereumChainConfigs: async () => {
+      await chainConfigStore.ready();
+      return chainConfigStore.getState().ethereumChainConfigs;
+    },
+    client: configureBackgroundTestClient(),
+    testnetMode: true,
+  }
+);
+
+export function getNetworksStore(
+  preferences: Pick<WalletRecord['publicPreferences'], 'testnetMode'>
+) {
+  return preferences.testnetMode?.on ? testenvNetworksStore : mainNetworksStore;
+}
+
+interface FetchNetworkOptions {
+  preferences: Pick<WalletRecord['publicPreferences'], 'testnetMode'>;
+  /**
+   * 'current' chooses networks store based on preferences.testnetMode value
+   * 'mainnet' uses mainnet networks store
+   * 'both' will try testnet networks store first IF {preferences.testnetMode.on}, if no network is found, it will try mainNetworksStore
+   */
+  apiEnv: 'mainnet' | 'both' | 'current';
```

Comment:
Maybe both can be changed to preferTestnet

Reply:
I think this name would have the same problem as "both": it may seem that testnet api will be always queried.
Other ideas? :)

Reply:
testnetFirst ?
Like cache-first in defi-sdk. Doesn't guarantee that cache exists

Reply:
Kinda works, thanks